app/src/main/java/org/simple/clinic/encounter/EncounterRepository.kt

```diff
@@ -32,7 +34,32 @@ class EncounterRepository @Inject constructor(
   }
 
   override fun mergeWithLocalData(payloads: List<EncounterPayload>): Completable {
-    TODO("not implemented")
+    return Single.fromCallable {
+      payloads
+          .filter { payload ->
+            database.encountersDao()
+                .getOne(payload.uuid)
+                ?.syncStatus.canBeOverriddenByServerCopy()
+          }
```

Comment:
I am not really sure about using a `filter` to run async operations. Can we use a `flatMap` to perform these queries and use a filter to just specify if the local copy be overridden by the server copy?

Reply:
I don't understand the issue in `filter` running in async ops. Can you explain this a little? 
 I will make the change anyway.

Reply:
Will this work?

```kotlin
  override fun mergeWithLocalData(payloads: List<EncounterPayload>): Completable {
    val encounterPayloads = Observable.fromIterable(payloads)
    val encounters = encounterPayloads.flatMap(::getEncounter)

    return Observables.zip(encounterPayloads, encounters)
        .filter { (_, encounter) -> encounter.isNotEmpty() }
        .map { (payload, encounter) -> payload to encounter.toNullable()!! }
        .filter { (_, encounter) -> encounter.syncStatus.canBeOverriddenByServerCopy() }
        .map { (payload, _) -> payloadToEncounters(payload) }
        .toList()
        .flatMapCompletable { saveMergedEncounters(it) }
  }

  private fun getEncounter(payload: EncounterPayload): Observable<Optional<Encounter>> =
      Observable.fromCallable { database.encountersDao().getOne(payload.uuid).toOptional() }
```